core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -881,6 +881,7 @@ class ReplicaManager(val config: KafkaConfig,
               if (partition.futureReplicaDirChanged(destinationDir)) {
                 replicaAlterLogDirsManager.removeFetcherForPartitions(Set(topicPartition))
                 partition.removeFutureLocalReplica()
+                logManager.resumeCleaning(topicPartition)
```

Comment:
In general, I agree with this change. In this case we should resume cleaning because we do not know when or whether the subsequent LeaderAndISR request will arrive. Is there a way you can test this with a unit test(s)?

Reply:
This statement is for canceling altering log dir (raising an alter log dir request whose future log dir is the current log dir). If we don't resume cleaning here, it will never be resumed.